File: web-app/api-server/models/Movie.js
const mongoose = require("mongoose");
const Schema = mongoose.Schema;

const movieSchema = new mongoose.Schema(
  {
    movieTitle: { type: String, default: "" },
    movieName: { type: String, default: "" },
    originalName: { type: String, default: "" },
    genreId: {
      type: String,
      default: `${Schema.Types.ObjectId}`,
      ref: "Genre",
    },
    releaseDate: { type: Date || null, default: null },
    firstAirDate: { type: Date || null, default: null },
    createdDate: { type: Date, default: Date.now() },
    updatedDate: { type: Date || null, default: null },
    countries: [
      {
        countryName: { type: String, default: "" },
        countryId: {
          type: String,
          default: `${Schema.Types.ObjectId}`,
          ref: "Country",
        },
      },
    ],
    originalLanguage: { type: String, default: "" },
    cateMovies: [
      {
        name: { type: String, default: "" },
        cateMovieId: {
          type: String,
          default: `${Schema.Types.ObjectId}`,
          ref: "CateMovie",
        },
      },
    ],
    overview: { type: String, default: "" },
    popularity: { type: Number, default: 0 },
    voteAverage: { type: Number, default: 0 },
    voteCount: { type: Number, default: 0 },
    video: { type: String, default: "" },
    trailer: { type: String, default: "" },
    isMovieAvailable: { type: Boolean, default: false },
    genre: { type: String, default: "" },
    isSeries: { type: Boolean, default: false },
  },
  { timestamps: true }
);

module.exports = mongoose.model("Movie", movieSchema);

File: frontend/mobile-app/ios-native/UIKit/MovaStream/MovaStream/Models/Movie.swift
//
//  Movie.swift
//  MovaStream
//
//  Created by HauNguyen on 27/04/2566 BE.
//

import Foundation

struct Movie: Codable {
    let title: String
    let backdrop_path: String
    let media_type: String?
    let release_date: String?
    let first_air_date: String?
    let genre_ids: [Double]
    let id: Double
    let name: String?
    let origin_country: [String]?
    let original_language: String
    let original_name: String?
    let overview: String
    let popularity: Double
    let poster_path: String?
    let vote_average: Double
    let vote_count: Double
}
